format/spec.md

```diff
@@ -444,6 +449,9 @@ Sorting floating-point numbers should produce the following behavior: `-NaN` < `
 
 A data or delete file is associated with a sort order by the sort order's id within [a manifest](#manifests). Therefore, the table must declare all the sort orders for lookup. A table could also be configured with a default sort order id, indicating how the new data should be sorted by default. Writers should use this default sort order to sort the data on write, but are not required to if the default order is prohibitively expensive, as it would be for streaming writes.
 
+Note:
+
+1. The ability to sort `variant` columns and the specific sort order is determined by the engines.
```

Comment:
Do we need this? I think anything we don't specify is up to engines already.

Reply:
OK. I will remove that then. Do we need to call out "Variant values cannot be present in an Iceberg sort order"?

Reply:
I think we should specifically forbid sort orders containing a variant. I think we actually are underdetermined in the spec here.

We have the following checks in the Reference Implementation
https://github.com/apache/iceberg/blob/8a16a417492eb6ccb1895b4b6db3536ff2a8b67d/api/src/main/java/org/apache/iceberg/SortOrder.java#L301-L311

So currently, even though we don't specify this here, you cannot make a sort order with array or map. I think we should explicitly call this out and add variant as well. My real concern here is that we add the ability to sort on something but don't define what that sorting actually looks like.